### PR TITLE
 Adding fit uncertainty exclusion from f-test, automatically rebin to remove zero bins and adjust fit max

### PR DESCRIPTION
**Pull request by OzAmram**
New fit options `--rebin` which automatically merges bins so chi2 calculation happens with enough stats and automatically adjusts end range of fit to match data. 
Now require that the fit fractional uncertainty evaluated at the signal  for a given number of parameters be less than some threshold to be included in f-test (default is 50%) configurable with option `--err_thresh` 
Added some other fit options (`--ftest_thresh`, `--mjj_max`) and also increased bin sizes used at high mass for chi2 computation so it was a bit more stable

### CODE REVIEW

File: fitting/draw_signals.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import ROOT
```
Comment (lmoureaux) [inline fitting/draw_signals.py]: Did you mean to include this file in the PR? If so it would be good to unhardcode `TNT` strings

File: fitting/interpolation.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 args = parser.parse_args()
 ### use same settings as in diboson search
-all_graphs = {"graviton" : "mean:spline,sigma:spline,alpha:pol1,sign:pol1,alpha2:pol2,sign2:pol2"}
+all_graphs = {"diboson" : "mean:spline,sigma:spline,alpha:pol1,sign:pol1,alpha2:pol2,sign2:pol2"}
```
Comment (lmoureaux) [inline fitting/interpolation.py]: This is gonna break our scripts, can we keep it unchanged?

File: fitting/Utils.py [hunk 3]
```diff
@@ -532,13 +554,13 @@ def checkSBFit(filename,label,roobins,plotname, nPars, plot_dir):
...
     pdf_names = ["model_s"] 
-    PlotFitResults(frame,fres.GetName(),nPars+1,frame3,"data_obs", pdf_names,chi2,ndof,'sbFit_'+plotname, plot_dir, has_sig = True)
+    PlotFitResults(frame,fres.GetName(),nPars+1,frame3,"data_obs", pdf_names,chi2,ndof,'sbFit_'+plotname, plot_dir, has_sig = True, plot_label = plot_label)
 
     print "chi2,ndof are", chi2, ndof
     return chi2, ndof
 
 
-def f_test(nParams, nDof, chi2, thresh = 0.05):
+def f_test(nParams, nDof, chi2, fit_errs, thresh = 0.05, err_thresh = 0.3 ):
```
Comment (lmoureaux) [inline fitting/Utils.py]: The default command line argument for `err_thresh` is 0.5. Let's not introduce default values in more than one place, or at the very least keep them consistent.

File: fitting/Utils.py [hunk 4]
```diff
@@ -400,11 +413,12 @@ def check_rough_sig(h_file, m_low, m_high):
...
         is_sig = f['truth_label'][()]
 
+    eps = 1e-6
     in_window = (mjj > m_low) & (mjj < m_high)
     sig_events = is_sig > 0.9
     bkg_events = is_sig < 0.1
-    S = mjj[sig_events & in_window].shape[0]
-    B = mjj[bkg_events & in_window].shape[0]
+    S = max(mjj[sig_events & in_window].shape[0], eps)
+    B = max(mjj[bkg_events & in_window].shape[0], eps)
```
Comment (lmoureaux) [inline fitting/Utils.py:421]: Good fix, I recall this crashing some of my fits

File: fitting/interpolation.py (hunk 2 above)
Comment (OzAmram) [reply fitting/interpolation.py]: Ok sure, I just got confused by the naming but I can change it back

File: fitting/draw_signals.py (hunk 1 above)
Comment (OzAmram) [reply fitting/draw_signals.py]: Yeah I thought it could be a useful script for others to make the same plot but I was also too lazy to make it fully generic haha. I can just remove it

File: fitting/Utils.py (hunk 3 above)
Comment (OzAmram) [reply fitting/Utils.py]: Ah yeah good catch. I'll change the default of this function to match at least. I want to keep a default to not break anyone else's who uses this function

**Comment by lmoureaux** [comment]
I think it's time to solve conflicts and merge this in `master` before it diverges too much

**Comment by OzAmram** [comment]
Yeah I agree. Just fixed the conflicts. 
We can merge it now unless anyone one objects
